Gems/Atom/RHI/DX12/Code/Source/RHI/StagingMemoryAllocator.cpp

```diff
@@ -27,6 +27,7 @@ namespace AZ
                 poolDesc.m_pageSizeInBytes = descriptor.m_mediumPageSizeInBytes;
                 poolDesc.m_collectLatency = descriptor.m_collectLatency;
                 poolDesc.m_getHeapMemoryUsageFunction = [this]() { return &m_memoryUsage; };
+                poolDesc.m_recycleOnCollect = true;
```

Comment:
Dont we need this for Vulkan and Metal too?

Reply:
Vulkan and Metal is using BufferPool which is hard-coded for all bufferPool instances. 
Although I didn't find similar allocations when running with Vulkan.

Reply:
For Vk/Metal we use m_stagingBufferPool and all BufferPools are getting hardcoded to 
`allocatorDescriptor.m_recycleOnCollect = false;`  within Vk/Metal::BufferPool::InitInternal. 

We could modify RHI::BufferPoolDescriptor to add m_recycleOnCollect which can be used to set that variable in BufferPool::InitInternal instead of always hardcoding it to false.